src/Components/index.tsx

```diff
@@ -1,18 +1,29 @@
-import Artwork from "./Artwork"
-import ArtworkGrid from "./ArtworkGrid"
-import BorderedPulldown from "./BorderedPulldown"
+// @ts-ignore - https://github.com/styled-components/styled-components/issues/1063#issuecomment-320344957
+import React from "react"
+// @ts-ignore
+import Artwork, { ArtworkProps } from "./Artwork"
+// @ts-ignore
+import ArtworkGrid, { ArtworkGridContainerProps } from "./ArtworkGrid"
+// @ts-ignore
+import BorderedPulldown, { BorderedPullDownProps } from "./BorderedPulldown"
 import Buttons from "./Buttons"
 import Checkbox from "./Checkbox"
-import Icon from "./Icon"
-import Input from "./Input"
+// @ts-ignore
+import Icon, { IconProps } from "./Icon"
+// @ts-ignore
+import Input, { InputProps } from "./Input"
 import Modal from "./Modal/Modal"
-import Nav from "./Nav"
-import Text from "./Text"
+// @ts-ignore
+import Nav, { NavBarProps } from "./Nav"
+// @ts-ignore
+import Text, { TextProps } from "./Text"
 import TextArea from "./TextArea"
-import TextLink from "./TextLink"
-import Title from "./Title"
+// @ts-ignore
+import TextLink, { LinkProps } from "./TextLink"
+// @ts-ignore
```

Comment:
@l2succes - I wonder if there is some way we could ignore imported interfaces without having to flag each unused import -- do you know if there is a TSLint / ts config option that would allow us to keep this clean?

Also, i'm pretty sure this is a shot in the dark, but any way to auto-export interfaces? 

@orta / @alloy - have y'all ever come across a good way to do this?

Reply:
@damassi unfortunately, I don't think there's a way to ignore a whole block 😔 (Related: https://github.com/Microsoft/TypeScript/issues/19573)

They just fixed the main issue that enables us to emit types in the first place with typescript@2.9.1 since this issue is related I think they should give us a way to auto-import interfaces when generating types

Reply:
Ahh cool cool, so I imagine we'll just have to track this as the feature becomes more mature 👍

Reply:
@alloy - in order for the types to be automatically emitted, it needs an in-scope reference to the interface  -- in the example above, `LinkProps` needs to be made available BUT because we're not using it directly in the code (only during the compilation process) it needs to be ignored due to the `noUnusedLocals` setting. 

This aspect of type emission kinda bugs me because it clutters up code with some developer vs compiler indirection, but I suspect now that this feature has landed there will be some overall tooling improvements coming soon that will help clean the process up.